src/components/scripts/oauth.mjs

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class OAuthConnector {
+  #redirectUri;
+  #clientId;
+  #clientSecret;
+  #oauthScopes = [];
+  #socketServer;
+  #authenticationCode;
+  #accessCode;
+  #refreshCode;
+  #promiseResolver;
+  #promiseRejecter;
+
+  #authentication_endpoint;
+  #token_endpoint;
+
+  constructor(authentication_endpoint, token_endpoint) {
+    this.#authentication_endpoint = authentication_endpoint;
+    this.#token_endpoint = token_endpoint;
+
+    this.#socketServer = new TCPServerSocket("::");
```

Comment:
Why `::`? You only need a localhost listener, am I getting this right?

Reply:
Correct. What would be the most appropriate syntax for this?

Reply:
If you need an Ipv4 listener (which is the most likely case), then `127.0.0.1` will do :) Otherwise `::1`

Reply:
Done.